manifests/repo.pp

```diff
@@ -114,8 +135,27 @@
         autorefresh  => $autorefresh_num,
       }
     }
+    'apt': {
+      apt::source { $name:
+        ensure         => $enabled_real,
+        location       => $baseurl_real,
+        comment        => $descr,
+        allow_unsigned => $gpgcheck_num,
+        key            => $gpgkey_hash,
+        repos          => $apt_repos,
+        release        => $apt_release,
+      }
+    }
     default: {
-      fail("Invalid repotype ${repotype}. Supported repotypes are yum, zypper and apt.")
+      fail("Invalid repotype ${repotype}. Supported repotypes are apt, yum and zypper.")
+    }
+  }
+
+  # Associate .repo files in directory for yum and zypper
```

Comment:
Did you try this and is this really needed ?
I think Puppet will not purge any file that is managed in the catalog.

Reply:
Hi. This is neccessary for the cases where APT is used. It appears it doesn't associate the files correctly with its parent directory if the files are defined in a separate module (apt) (if I remember correctly). Without this code it put the repository files into the directory and then removed them the next run (if purge is active).

Reply:
Strange, I did expect Puppet to respect ALL files that are managed in the catalog to not get purged.